UI/SitRepPanel.cpp

```diff
@@ -494,68 +494,107 @@ namespace {
             for (auto& entry : empires)
                 sr_empires.insert(entry.second);
         }
+
+        std::map<int, std::list<SitRepEntry>> turns;
         for (auto sitrep_empire : sr_empires) {
-            for (Empire::SitRepItr sitrep_it = sitrep_empire->SitRepBegin(); sitrep_it != sitrep_empire->SitRepEnd(); ++sitrep_it) {
-                if (!verbose_sitrep) {
-                    if (!sitrep_it->Validate())
-                        continue;
-                }
-                if (hidden_sitrep_templates.find(sitrep_it->GetLabelString().empty() ? sitrep_it->GetTemplateString() : sitrep_it->GetLabelString()) != hidden_sitrep_templates.end())
-                    continue;
-                if (permanently_snoozed_sitreps.find(sitrep_it->GetText()) != permanently_snoozed_sitreps.end())
-                    continue;
-                auto sitrep_set_it = snoozed_sitreps.find(sitrep_it->GetTurn());
-                if (sitrep_set_it != snoozed_sitreps.end() && sitrep_set_it->second.find(sitrep_it->GetText()) != sitrep_set_it->second.end())
-                    continue;
+            for (auto sitrep_it = sitrep_empire->SitRepBegin(); sitrep_it != sitrep_empire->SitRepEnd(); ++sitrep_it) {
                 turns[sitrep_it->GetTurn()].push_back(*sitrep_it);
             }
         }
         return turns;
     }
 }
 
-int SitRepPanel::GetNextNonEmptySitrepsTurn(const std::map<int, std::list<SitRepEntry>>& turns, int turn, bool forward) const {
+/** Return true if the sitrep is not hidden, validates and is not snoozed. */
+bool SitRepPanel::IsSitRepInvalid(SitRepEntry& sitrep) const {
+    if (m_hidden_sitrep_templates.count(
+            sitrep.GetLabelString().empty() ? sitrep.GetTemplateString() : sitrep.GetLabelString()))
+    { return true; }
+
+    // Validation is time consuming because all variables are substituted
+    bool validated = sitrep.Validate();
+
+    // verbose-sitrep will hide all sitreps that do not validate
```

Comment:
seems misleading... I'd say having verbose-sitrep off / disabled will hide sitreps that do not validate

Reply:
fixed.